java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -567,6 +569,9 @@ public void visitVariable(VariableTree tree) {
     //skip type, it has been resolved in second pass
     if (tree.initializer() != null) {
```

Comment:
Would be clearer to use a variable for the initializer instead of calling this method 4 times.